packages/ui/src/components/Catalog/BaseCatalog.tsx

```diff
@@ -28,8 +29,12 @@ export const BaseCatalog: FunctionComponent<BaseCatalogProps> = (props) => {
     [onTileClick, props.tiles],
   );
 
+  useEffect(() => {
+    document.getElementById('catalog-list')!.scrollTop = 0;
+  }, [props.tiles]);
+
```

Comment:
@tplevko This is a great addition 🎉 , I would just suggest if possible, using `Ref` to get the element instead.

Maybe something like this could work:

```tsx
  const catalogBodyRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    catalogBodyRef.current?.scrollTop = 0;
  }, [props.tiles]);

  return (
    <div id="catalog-list" className="catalog-list" ref={catalogBodyRef}>
  ...
```

This way we're resilient against changes and more important, we delegate the object fetching to react.

Reply:
Great, thanks for the hint! Works with no issue.

I also moved the div, so we would have static displayed element counter at the top.

https://github.com/KaotoIO/kaoto-next/assets/4180208/d604d97e-d1d9-4f6b-a68d-49df28cfafbe